includes/embeds/class-amp-youtube-embed-handler.php

```diff
@@ -236,6 +444,51 @@ private function get_video_id_from_url( $url ) {
 		return false;
 	}
 
+	/**
+	 * Get the start time of the YouTube video in seconds.
+	 *
+	 * @param string $url YouTube URL.
+	 *
+	 * @return int Start time in seconds.
+	 */
+	private function get_start_time_from_url( $url ) {
+
+		if ( empty( $url ) ) {
+			return 0;
+		}
+
+		$start_time = 0;
+		$parsed_url = wp_parse_url( $url );
+
+		if ( ! empty( $parsed_url['query'] ) ) {
+			$query_vars = [];
+			wp_parse_str( $parsed_url['query'], $query_vars );
+
+			if ( ! empty( $query_vars['start'] ) && 0 < (int) $query_vars['start'] ) {
+				return (int) $query_vars['start'];
+			}
+		}
+
+		if ( ! empty( $parsed_url['fragment'] ) ) {
+			$regex = '/^t=(?<minutes>\d+)m(?<seconds>\d+)s$/';
```

Comment:
Actually the `t` fragment turns out to be more complex than expected. All of these are valid:

https://www.youtube.com/watch?v=kfVsfOSbJY0#t=62   # 62 secs
https://www.youtube.com/watch?v=kfVsfOSbJY0#t=1m2s # 1 min 2 secs
https://www.youtube.com/watch?v=kfVsfOSbJY0#t=62s  # 62 secs
https://www.youtube.com/watch?v=kfVsfOSbJY0#t=1m   # 1 min
https://www.youtube.com/watch?v=kfVsfOSbJY0#t=1m2  # 1 mins 2 secs

So that complicates the regex a bit. This should do:

```suggestion
			$regex = '/^t=(?:(?<minutes>\d+)m)?(?:(?<seconds>\d+)s?)?$/';
```